doc/source/whatsnew/v0.21.0.txt

```diff
@@ -283,6 +283,7 @@ Other API Changes
 - The Categorical constructor no longer accepts a scalar for the ``categories`` keyword. (:issue:`16022`)
 - Accessing a non-existent attribute on a closed :class:`~pandas.HDFStore` will now
   raise an ``AttributeError`` rather than a ``ClosedFileError`` (:issue:`16301`)
+- :func:`read_csv` now issues a ``UserWarning`` if the ``names`` parameter contains duplicates (:issue:`17095`)
```

Comment:
should be FutureWarning

Reply:
Doh!  My bad for not catching that.  Fixed.

Reply:
Changing back to `UserWarning` in light of later discussion.